searchcore/src/vespa/searchcore/proton/server/rpc_hooks.cpp

```diff
@@ -184,18 +184,19 @@ RPCHooksBase::initRPC()
 }
 
 RPCHooksBase::Params::Params(Proton &parent, uint32_t port, const vespalib::string &ident)
-        : proton(parent),
-          slobrok_config(config::ConfigUri("admin/slobrok.0")),
-          identity(ident),
-          rtcPort(port)
+    : proton(parent),
+      slobrok_config(config::ConfigUri("admin/slobrok.0")),
+      identity(ident),
+      rtcPort(port)
 { }
 
 RPCHooksBase::Params::~Params() = default;
 
 RPCHooksBase::RPCHooksBase(Params &params)
     : _proton(params.proton),
-      _docsumByRPC(new DocsumByRPC(_proton.getDocsumBySlime())),
-      _orb(std::make_unique<FRT_Supervisor>()),
+      _docsumByRPC(std::make_unique<DocsumByRPC>(_proton.getDocsumBySlime())),
+      _transport(std::make_unique<FNET_Transport>(2)),
```

Comment:
when using a separate Transport, its lifetime needs to be handled separately. It also makes the supervisor crash when calling functions that expects it to totally control its own transport. (Start/ShutDown/WaitFinished)

Reply:
Now I think it should be better.